plan/plans.go

```diff
@@ -70,13 +73,87 @@ type Prepare struct {
 	SQLText string
 }
 
+// Prepared represents a prepared statement.
+type Prepared struct {
+	Stmt          ast.StmtNode
+	Params        []*ast.ParamMarkerExpr
+	SchemaVersion int64
+	UseCache      bool
+}
+
 // Execute represents prepare plan.
 type Execute struct {
 	basePlan
 
 	Name      string
 	UsingVars []expression.Expression
```

Comment:
Why the using vars is expression but datum ?

Reply:
It may be user variable, need to be evaluated first.

Reply:
ok